### PR TITLE
bakcport fixes to 0.6.1

### PR DESCRIPTION
**Pull request by psychocoderHPC**
Backoport:

- fix CPU static shared memory implementation #1258
- units tests: fix queue test #1266
- document return value of `empty()` and `isComplete()` #1265
- update changelog

### CODE REVIEW

**Review by bernhardmgruber — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,9 +7,12 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 ### Bug Fixes:
 - CI: use ubuntu-18.04 for gcc-5 and gcc-6 builds #1252
 - fix OpenMP 5 shared memory allocation #1254
+- fix CPU static shared memory implementation #1258
+- units tests: fix queue test #1266
```
Comment (sbastrakov) [inline CHANGELOG.md]: Typo: should be `unit`, not `units`. Not sure if a big deal, but in case it is a public changelog worth fixing I think.

Comment (sbastrakov) [reply CHANGELOG.md]: Forgot to ping @psychocoderHPC

Comment (psychocoderHPC) [reply CHANGELOG.md]: Well spotted, I will fix it. Therefore we have the review process. No need to ping, the PR author will by default be notified about any comments.

Comment (psychocoderHPC) [reply CHANGELOG.md]: fixed

**Comment by bernhardmgruber** [comment]
@sbastrakov if you are fine, we can hit the merge button.

**Review by sbastrakov — APPROVED** [review]

**Pull request merged**